### PR TITLE
doc/start: add testing support information

### PR DESCRIPTION
**Pull request by zdover23**
This PR adds information about support for testing,
and information about which distros the Ceph project
builds packages for.

This is one in a series of PRs including the following:

https://github.com/ceph/ceph/pull/45385
https://github.com/ceph/ceph/pull/45764

This PR specifically includes the information that Ernesto
Puerta collected here:
https://github.com/ceph/ceph/pull/45385#pullrequestreview-911766656

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/start/os-recommendations.rst [hunk 1]
```diff
@@ -38,19 +38,27 @@ Platforms
...
-+--------------+--------+--------+---------------+----------+--------+
-| Pacific      | 16.2.z | 8      | 18.04, 20.04  | 15.2     | 10, 11 |
-+--------------+--------+--------+---------------+----------+--------+
-| Octopus      | 15.2.z | 7, 8   | 18.04, 20.04  | 15.2     | 10     |
-+--------------+--------+--------+---------------+----------+--------+
-
++--------------+--------+------------------------+---------------+----------+-----------------+
+| Release Name | Tag    | CentOS                 | Ubuntu        | OpenSUSE | Debian          |
++==============+========+========================+===============+==========+=================+
+| Quincy       | 17.2.z | 8 :sup:`A`             | 20.04         | 15.3     | 11 :sup:`C`     |
```
Comment (jdurgin) [inline doc/start/os-recommendations.rst:47]: A  applies to 20.04 for each ceph release. 18.04 is C at this point for pacific and octopus.

C applies to all opensuse and debian versions - maybe superscript the column header for those?

File: doc/start/os-recommendations.rst [hunk 2]
```diff
@@ -38,19 +38,27 @@ Platforms
...
+- **A**: Ceph provides packages and has done comprehensive tests on the software in them.
+- **B**: Ceph provides packages and has done basic tests on the software in them.
+- **C**: Ceph provides packages only. No tests have been done on these releases.
+
+.. note::
+   **For Centos 7 Users** 
+   
+   ``Btrfs`` is no longer tested on Centos 7 in the Octopus release. We recommend using ``bluestore`` instead.
 
 Notes
```
Comment (jdurgin) [inline doc/start/os-recommendations.rst:63]: looks like we can get rid of the rest of this notes section now as well

Comment (zdover23) [reply doc/start/os-recommendations.rst:63]: Yeah, that's the next commit that waits on this one to merge.

File: doc/start/os-recommendations.rst (hunk 1 above)
Comment (zdover23) [reply doc/start/os-recommendations.rst]: I've shifted the column around so that it looks like this.

**Review by jdurgin — APPROVED** [review]

**Pull request merged**